lib/conjur/debify.rb

```diff
@@ -244,22 +244,51 @@ def ignore_file? f
     fpm_image = Docker::Image.build_from_dir File.expand_path('fpm', File.dirname(__FILE__)), tag: "debify-fpm", &DebugMixin::DOCKER
     DebugMixin.debug_write "Built base fpm image '#{fpm_image.id}'\n"
     dir = File.expand_path(dir)
+    # puts "DIR ----------------"######
+    # puts dir ########
+    # puts pwd
     Dir.chdir dir do
       version = cmd_options[:version] || detect_version
+      # dockerfile_path = cmd_options[:dockerfile] || File.expand_path("debify/Dockerfile.fpm", pwd)
+      # dockerfile = File.read(dockerfile_path)
+
+      # output = StringIO.new
+      # Gem::Package::TarWriter.new(output) do |tar|
+      #   git_files.each do |fname| #################################################### <- safe_files
+      #     stat = File.stat(fname)
+      #     tar.add_file(fname, stat.mode) { |tar_file| tar_file.write(File.read(fname)) }
+      #   end
+      #   tar.add_file('Dockerfile', 0640) { |tar_file| tar_file.write dockerfile.gsub("@@image@@", fpm_image.id) }
+      # end
+      # output.rewind
+
+      # # puts "Dockerfile----------------------" #############################
+      # # puts dockerfile
+      # # puts "Git Files--------------------------"
+      # # puts git_files
+      # # puts "Output---------------------" #################################
+      # # puts output.string
+
+      # image = Docker::Image.build_from_tar output, &DebugMixin::DOCKER
+
+      ## debug
+      puts dir
+
+      # change image name in Dockerfile before being turned into tar
       dockerfile_path = cmd_options[:dockerfile] || File.expand_path("debify/Dockerfile.fpm", pwd)
       dockerfile = File.read(dockerfile_path)
+      replace_image = dockerfile.gsub("@@image@@", fpm_image.id)
+      File.open(dockerfile_path, "w") {|file| file.puts replace_image}
 
-      output = StringIO.new
-      Gem::Package::TarWriter.new(output) do |tar|
-        git_files.each do |fname|
-          stat = File.stat(fname)
-          tar.add_file(fname, stat.mode) { |tar_file| tar_file.write(File.read(fname)) }
-        end
-        tar.add_file('Dockerfile', 0640) { |tar_file| tar_file.write dockerfile.gsub("@@image@@", fpm_image.id) }
-      end
-      output.rewind
+      # build image from project being debified dir
+      image = Docker::Image.build_from_dir dir, &DebugMixin::DOCKER
 
-      image = Docker::Image.build_from_tar output, &DebugMixin::DOCKER
```

Comment:
This change does not specify the Dockerfile to be used, so it will always try to use `Dockerfile`, even though you've replaced `@@image@@` in what is defined in `dockerfile_path`.  Note that when it does `build_from_tar`, it writes the filename as `Dockerfile` inside the tar file, so it doesn't need to be specified in the `build_from_tar` function.

Reply:
I'm still using the same login of defining the `dockerfile_path`. Do you think adding a line to name whatever is at the end of `dockerfile_path` to `Dockerfile` would be enough to fix that?

Reply:
`Dockerfile` may already exist in the directory and would be overwritten with unrelated content (since the `dockerfile_path` will typically be `Dockerfile.fpm`).  A quick look didn't show a way to pass in the file to use as the Dockerfile to `build_from_dir`, but looking at the docker API itself, it looks like you can set `dockerfile` to `dockerfile_path` by changing it to:

```suggestion
      image = Docker::Image.build_from_dir dir, dockerfile: dockerfile_path, &DebugMixin::DOCKER
```

I would still suggest using a clean directory approach rather than trying to assemble it on top of the existing directory, though.